plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotMetadata.java

```diff
@@ -303,8 +305,11 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
             Map<ColumnHandle, Domain> supported = new HashMap<>();
             Map<ColumnHandle, Domain> unsupported = new HashMap<>();
             for (Map.Entry<ColumnHandle, Domain> entry : domains.entrySet()) {
+                Type type = ((PinotColumnHandle) entry.getKey()).getDataType();
+                // Do not push down filters for arrays or varbinary columns:
                 // Pinot does not support array literals
-                if (((PinotColumnHandle) entry.getKey()).getDataType() instanceof ArrayType) {
+                // Pinot does not fully support varbinary null literals (empty byte array): https://github.com/apache/pinot/issues/4230
```

Comment:
Can we have a test detect this when this gets fixed on Pinot side? e.g. asserting that incorrect results are returned for filter on a varbinary columns with an actually empty array and a null value?